contracts/interfaces/modules/metadata/ICoreMetadataModule.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.23;
+
+import { IModule } from "../../../../contracts/interfaces/modules/base/IModule.sol";
+
+/// @title CoreMetadataModule
+/// @notice Manages the core metadata for IP assets within the Story Protocol.
+/// @dev This contract allows setting and updating core metadata attributes for IP assets.
+interface ICoreMetadataModule is IModule {
+    /// @notice Emitted when the nftTokenURI for an IP asset is set.
+    event NFTTokenURISet(address indexed ipId, string nftTokenURI, bytes32 nftMetadataHash);
+
+    /// @notice Emitted when the metadataURI for an IP asset is set.
+    event MetadataURISet(address indexed ipId, string metadataURI, bytes32 metadataHash);
+
+    /// @notice Emitted when all metadata for an IP asset are frozen.
+    event MetadataFrozen(address indexed ipId);
+
+    /// @notice Update the nftTokenURI for an IP asset,
+    /// by retrieve the latest TokenURI from IP NFT to which the IP Asset bound.
+    /// @dev Can only be called once per IP asset to prevent overwriting.
```

Comment:
Do you mean this setter can be called until metadata is frozen? So you can call as many times as wanted until frozen. Same question for @dev in the other setter functions.

Reply:
Right, the comments need to be updated. good catch! fixed.